src/hw/connectApp.js

```diff
@@ -59,15 +65,20 @@ const openAppFromDashboard = (
   appName
 ): Observable<ConnectAppEvent> =>
   concat(
-    of({ type: "device-permission-requested", wording: appName }),
+    of({ type: "ask-open-app", appName }),
     defer(() => from(openApp(transport, appName))).pipe(
       concatMap(() => of({ type: "device-permission-granted" })),
       catchError((e) => {
         if (e && e instanceof TransportStatusError) {
           switch (e.statusCode) {
             case 0x6984:
             case 0x6807:
-              return of({ type: "app-not-installed", appName });
+              return getEnv("EXPERIMENTAL_INLINE_INSTALL")
+                ? concat(
+                    streamAppInstall(transport, appName),
+                    from(openAppFromDashboard(transport, appName))
+                  )
```

Comment:
sorry to add this requirement at last minute. but when I helped you on this, i noticed that an error thrown here would still "loop forever".
We must make sure throwing an error from there would be displayed to either because we can't afford having an infinite loop of hsm list apps when something is bad with our server for instance.

we must make sure we let the error flow and that it interrupts the upstream polling somehow.
IMO it's even a blocker for the "experimental" mode because if many users starts using that it's still bad for the load on our api.

Reply:
As we talked on slack, I was unable to reach the endless loop when triggering an error inside the inlineInstall or the reducer. While there is still a chance it will happen, I am unable to reproduce, so if it's good enough to test (we could just not add it as an exposed experimental on the UI and make it only available if launched with the env for now) we could merge @gre